e2e/single-cluster/finalizers_test.go

```diff
@@ -0,0 +1,296 @@
+package singlecluster_test
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/fleet/e2e/testenv"
+	"github.com/rancher/fleet/e2e/testenv/kubectl"
+)
+
+var _ = Describe("Deleting a resource with finalizers", func() {
+	var (
+		k               kubectl.Command
+		gitrepoName     string
+		path            string
+		r               = rand.New(rand.NewSource(GinkgoRandomSeed()))
+		targetNamespace string
+	)
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		targetNamespace = testenv.NewNamespaceName("target", r)
+	})
+
+	JustBeforeEach(func() {
+		gitrepoName = testenv.RandomFilename("finalizers-test", r)
+		path = "simple-chart"
+	})
+
+	AfterEach(func() {
+		_, err := k.Namespace("cattle-fleet-system").Run(
+			"scale",
+			"deployment",
+			"fleet-controller",
+			"--replicas=1",
+			"--timeout=5s",
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, _ = k.Delete("gitrepo", gitrepoName)
+		_, _ = k.Delete("bundle", fmt.Sprintf("%s-%s", gitrepoName, path))
+		bdNamespace, err := k.Get(
+			"ns",
+			"-o=jsonpath='{.items[?(@."+
+				`metadata.annotations.fleet\.cattle\.io/cluster-namespace=="fleet-local"`+
+				")].metadata.name}'",
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, _ = env.Kubectl.Namespace(bdNamespace).Delete(
```

Comment:
Why do we need to "manually" delete the `bundledeployment`?
It should be gone after deleting the `bundle`, right?

In fact this should be part of the test... checking that when deleting a `bundle` or `gitrepo` the `bundledeployments` related are also gone

Reply:
I have removed this, as the bundle deployment will indeed have been deleted by each test case, either manually or through previous cleanup steps, ie scaling the Fleet controller back up, which would trigger `GitRepo` and bundle deletion.